commit 607f969f5dca2dc100cbc53e24ab49ac24ef8987
Author: Adam Dinwoodie <me-and@users.noreply.github.com>
Date:   Wed Jun 26 17:14:48 2024 +0100

    systemd: don't try to restart templates
    
    If the user has template services in their systemd configuration, these
    can't be restarted, and will produces warnings during the activation
    phase.  Avoid those warnings by skipping any uninstantiated templates
    when looking for services to start or stop.

diff --git a/modules/systemd-activate.sh b/modules/systemd-activate.sh
index 235f1a24..ca26fea8 100644
--- a/modules/systemd-activate.sh
+++ b/modules/systemd-activate.sh
@@ -34,16 +34,16 @@ function systemdPostReload() {
     touch "$oldServiceFiles"
   else
     find "$oldUserServicePath" \
-      -maxdepth 1 -name '*.service' -exec basename '{}' ';' \
+      -maxdepth 1 -name '*.service' \! -name '*@.service' -exec basename '{}' ';' \
       | sort \
       > "$oldServiceFiles"
   fi
 
   if [[ ! -d "$newUserServicePath" ]]; then
     touch "$newServiceFiles"
   else
     find "$newUserServicePath" \
-      -maxdepth 1 -name '*.service' -exec basename '{}' ';' \
+      -maxdepth 1 -name '*.service' \! -name '*@.service' -exec basename '{}' ';' \
       | sort \
       > "$newServiceFiles"
   fi
